File: src/lecture1/lecture1_topic_py/scripts/publisher.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import Int32

def publisher():
    rospy.init_node("publisher")
    p = rospy.Publisher("topic", Int32, queue_size=5)
    rate = rospy.Rate(10)
    c = 0
    while not rospy.is_shutdown():
        p.publish(c)
        c += 1
        rate.sleep()

if __name__ == '__main__':
    try:
        publisher()
    except rospy.ROSInterruptException:
        pass

File: src/lecture1/lecture1_topic_cpp/src/publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "publisher");
    ros::NodeHandle n;
    ros::Publisher p = n.advertise<std_msgs::Int32>("topic", 5);
    ros::Rate rate(10);
    int c = 0;
    while (ros::ok())
    {
        std_msgs::Int32 msg;
        msg.data = c++;
        p.publish(msg);
        rate.sleep();
    }
    return 0;
}

